src/NServiceBus.Core/Logging/ColoredConsoleLogger.cs

```diff
@@ -1,26 +1,23 @@
 ﻿namespace NServiceBus
 {
     using System;
-    using System.Runtime.InteropServices;
+    using System.IO;
     using Logging;
 
     static class ColoredConsoleLogger
     {
         static ColoredConsoleLogger()
         {
-            const int STD_OUTPUT_HANDLE = -11;
-            logToConsole = GetStdHandle(STD_OUTPUT_HANDLE) != IntPtr.Zero;
+            logToConsole = Console.OpenStandardOutput() != Stream.Null;
```

Comment:
It seems [`OpenStandardOutput` creates `__ConsoleStream`](https://referencesource.microsoft.com/#mscorlib/system/console.cs,507) which overrides `Dispose` and only sets the handle to null. Would you also agree that this way you wrote here is safe by not disposing the newly acquired ConsoleStream?